pkg/kafka/auth/config.go

```diff
@@ -86,6 +86,13 @@ func (config *AuthenticationConfig) InitFromViper(configPrefix string, v *viper.
 
 	config.TLS = tlsClientConfig.InitFromViper(v)
 
+	if config.Authentication == tls {
+		config.TLS.Enabled = true
+	}
+	if config.TLS.Enabled == true {
```

Comment:
in L83, can we change ShowEnabled to false?

Reply:
It will break jaeger-operator and clients which adopted 1.17

Reply:
Fair enough, but do we really want to keep this duality? We can extend the config to render tls.enabled as deprecated, and add warnings when used.

Reply:
I can try to do that, I don't like the current design either.

Reply:
I thought @jpkrohling had mentioned a required use case where TLS could be enabled, but authentication set to `none` or something other than `tls`?

Reply:
I have created an issue to remove them in the next release https://github.com/jaegertracing/jaeger/issues/2111

Reply:
@objectiser yes, the current flags are misleading for people because there is overlap betwen:

```
--kafka.consumer.authentication
--kafka.consumer.tls.enabled
```